src/libs/ReportSecondaryActionUtils.ts

```diff
@@ -0,0 +1,330 @@
+
+function isSubmitAction(report: Report, policy: Policy): boolean {
+    const isExpense = isExpenseReport(report);
+
+    if (!isExpense) {
+        return false;
+    }
+
+    const isSubmitter = isCurrentUserSubmitter(report.reportID);
+    const isApprover = isApprovedMember(policy, getCurrentUserAccountID());
+
+    if (!isSubmitter && !isApprover) {
+        return false;
+    }
+
+    const isOpen = isOpenReport(report);
+
+    if (!isOpen) {
+        return false;
+    }
+
+    const autoReportingFrequency = getCorrectedAutoReportingFrequency(policy);
+
+    const isScheduledSubmitEnabled = policy?.harvesting?.enabled && autoReportingFrequency !== CONST.POLICY.AUTO_REPORTING_FREQUENCIES.MANUAL;
+
+    return !!isScheduledSubmitEnabled;
+}
+
+function isApproveAction(report: Report, policy: Policy, reportTransactions: Transaction[], violations: OnyxCollection<TransactionViolation[]>): boolean {
+    const isExpense = isExpenseReport(report);
+    const isApprover = isApprovedMember(policy, getCurrentUserAccountID());
+    const isProcessing = isProcessingReport(report);
+    const hasDuplicates = reportTransactions.some((transaction) => isDuplicate(transaction.transactionID));
+
+    if (isExpense && isApprover && isProcessing && hasDuplicates) {
+        return true;
+    }
+
+    const transactionIDs = reportTransactions.map((t) => t.transactionID);
+
+    const hasAllPendingRTERViolations = allHavePendingRTERViolation(transactionIDs, violations);
+
+    if (hasAllPendingRTERViolations) {
+        return true;
+    }
+
+    const isAdmin = policy?.role === CONST.POLICY.ROLE.ADMIN;
+
+    const shouldShowBrokenConnectionViolation = shouldShowBrokenConnectionViolationForMultipleTransactions(transactionIDs, report, policy, violations);
+
+    const userControllsReport = isApprover || isAdmin;
+    return userControllsReport && shouldShowBrokenConnectionViolation;
+}
+
+function isUnapproveAction(report: Report, policy: Policy): boolean {
+    const isExpense = isExpenseReport(report);
+    const isApprover = isApprovedMember(policy, getCurrentUserAccountID());
+    const isApproved = isReportApproved({report});
+
+    return isExpense && isApprover && isApproved;
+}
+
+function isCancelPaymentAction(report: Report): boolean {
+    const isExpense = isExpenseReport(report);
+
+    if (!isExpense) {
+        return false;
+    }
+
+    const isPaidElsewhere = report.stateNum === CONST.REPORT.STATE_NUM.APPROVED && report.statusNum === CONST.REPORT.STATUS_NUM.REIMBURSED;
+
+    if (isPaidElsewhere) {
+        return true;
+    }
+
+    const isPaymentProcessing = true; // TODO
+    const hasDailyNachaCutoffPassed = false; // TODO
+    return isPaymentProcessing && !hasDailyNachaCutoffPassed;
+}
+
+function isExportAction(report: Report, policy: Policy): boolean {
+    const isInvoice = isInvoiceReport(report);
+    const isSender = true; // TODO is sender the same as submitter?
```

Comment:
@luacmartins I'm not sure how to determine this value

Reply:
Yes, sender (invoice) and submitter (expense reports) are the same.